packages/account-v2/src/utils/pooUtils.ts

```diff
@@ -30,3 +38,26 @@ export const generatePOOInitialValues = (paymentMethodData: TPaymentMethodData)
         return acc;
     }, {} as TProofOfOwnershipFormValue);
 };
+
+const maskCardNumber = (cardNumber: string) => {
+    if (
+        cardNumber.length !== 16 ||
+        (cardNumber.length === 16 && ValidationConstants.patterns.invalidFormattedCardNumberCharacters.test(cardNumber))
+    ) {
+        return cardNumber;
+    }
+    return `${cardNumber.substring(0, 6)}XXXXXX${cardNumber.substring(12)}`;
+};
+
+export const formatIdentifier = (identifierType: TPaymentMethodIdentifier, paymentMethodIdentifier?: string) => {
+    const formattedId = paymentMethodIdentifier?.replace(/\s/g, '') ?? '';
+    if (identifierType === PAYMENT_METHOD_IDENTIFIER.cardNumber) {
+        return maskCardNumber(formattedId);
+    } else if (
+        PAYMENT_METHOD_IDENTIFIER.email === identifierType ||
+        PAYMENT_METHOD_IDENTIFIER.userID === identifierType
```

Comment:
[PAYMENT_METHOD_IDENTIFIER.email,PAYMENT_METHOD_IDENTIFIER.userID] .includes(identifierType)

Reply:
This throws TS error saying that identifierType has to be string. Hence I had to split it into `||`